### ISSUE TITLE
Support currency conversion

### ISSUE THREAD

**Issue by samfreedman** [mattjgalloway/cgtcalc#14]
It would be great if cgtcalc supported automatic currency conversions. My understanding is that, for CGT purposes, any acquisitions of assets in non-sterling currency (eg stock bought in USD) should be calculated in GBP using the exchange rate at the time of acquisition (or the HMRC monthly average rate, or any other equivalent rate as long as you are consistent - I'll happily be pointed at a more accurate explanation though). Equally, disposals should be converted using the same rate from the originating currency back to GBP. This is important to correctly determine the Section 104 pool value.

I've been experimenting with cgtcalc, and have resorted to manually looking up the exchange rate on https://www.xe.com/currencytables/, and doing the conversion maths myself. It would be very helpful if the calculator parsed the provided `<AMOUNT>` in each row to determine what currency it was in, and used the transaction date to look up the historical conversion rate.

**Comment by mattjgalloway** [comment]
Yes, this would be nice. However would need to assume there's a network connection present, and fetch via some API or screen scraping. From what I've found, there's no decent APIs that are free, and screen scraping I'm not particularly up for maintaining that as it will change from time to time. Also generally not keen on requiring an internet connection.

An alternative would be users providing a table with currency conversions (date and conversion factor). Which is better, but would require people to create such a table.

Personally what I do here, is I keep all my records in GBP, and do the conversion around the time I do the transaction - usually once I've got the trade confirmation. And then I have all my records in GBP anyway. That's what I would recommend others to do.

**Comment by deed02392** [comment]
Here's a Python implementation using the free API, provided by HMRC themselves: https://github.com/KapJI/capital-gains-calculator/blob/main/cgt_calc/currency_converter.py

It actually generates a local cache of exchange rates, using monthly averages.

**Comment by romk1n** [comment]
that would be awesome to have an option to include different currencies and even just the conversion rate into pounds, thank you.

**Comment by matttan90** [comment]
Can I suggest an optional additional value <exchange rate> in the text file that we can use to multiply with to get GBP? And if not present, we just multiply 1 (ie, it's already in GBP).

That decouples us from fetching from APIs, and you still keep all the raw data (ie, share prices in USD or other currencies, and exchange rates that we used).

**Comment by mattjgalloway** [comment]
> Can I suggest an optional additional value in the text file that we can use to multiply with to get GBP? And if not present, we just multiply 1 (ie, it's already in GBP).
> 
> That decouples us from fetching from APIs, and you still keep all the raw data (ie, share prices in USD or other currencies, and exchange rates that we used).

It's a good idea, but I still generally stand by my view that the calculator should be dealing with one thing only and not bringing in any extra complexity. You could easily write a script to use such a text file and output the right format for cgtcalc for example.

Though - I'll have a think about supplying such a script as either a standalone tool or a sub-tool of cgtcalc. I could get behind that as a good compromise.

**Comment by mattjgalloway** [comment]
I'm not going to support currency conversion sadly. I don't think it has a place in this part of calculation. My view is you should keep your records in GBP always and then feed into the calculator in GBP.

There's space for a layer above the calculator that would take in data from your brokers and apply currency conversion to spit out cgtcalc's format. However I would say that should be a separate project.
